be/src/exec/spill/input_stream.cpp

```diff
@@ -476,13 +476,13 @@ StatusOr<InputStreamPtr> BlockGroupSet::build_ordered_stream(std::vector<BlockGr
                                                              RuntimeState* state, const SerdePtr& serde,
                                                              Spiller* spiller, const SortExecExprs* sort_exprs,
                                                              const SortDescs* sort_descs) {
+    BlockReaderOptions read_options;
+    if (spiller->options().enable_buffer_read) {
+        read_options.enable_buffer_read = true;
+        read_options.max_buffer_bytes = spiller->options().max_read_buffer_bytes / block_groups.size();
```

Comment:
```suggestion
        read_options.max_buffer_bytes = spiller->options().max_read_buffer_bytes / block_groups.size();
```
I think you need to add some protection code to make sure that `read_options.max_buffer_bytes` is in a reasonable range.